xapian-core/net/remoteserver.cc

```diff
@@ -968,3 +983,83 @@ RemoteServer::msg_removespelling(const string & message)
     pack_uint_last(reply, wdb->remove_spelling(string(p, p_end - p), freqdec));
     send_message(REPLY_REMOVESPELLING, reply);
 }
+
+void
+RemoteServer::msg_getsynonymtermlist(const string& message)
+{
+
+    Xapian::TermIterator t = db->synonyms_begin(message);
+    string reply, prev;
+    while (t != db->synonyms_end(message)) {
+	if (rare(prev.size() > 255))
+	    prev.resize(255);
+	const string& term = *t;
+	size_t reuse = common_prefix_length(prev, term);
+	reply.append(1, char(reuse));
+	pack_uint(reply, term.size() - reuse);
+	reply.append(term, reuse, string::npos);
+	prev = term;
+	++t;
+    }
+    send_message(REPLY_SYNONYMTERMLIST, reply);
+}
+
+void
+RemoteServer::msg_getsynonymkeylist(const string& message)
+{
+    Xapian::TermIterator t = db->synonym_keys_begin(message);
+    string reply, prev;
```

Comment:
And here `prev` should be initialised to `message`, like in `RemoteServer::msg_metadatakeylist()`.